src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -50,9 +50,22 @@ public final class InlineConfigParser {
     /** A pattern matching the symbol: "\" or "/". */
     private static final Pattern SLASH_PATTERN = Pattern.compile("[\\\\/]");
 
-    /** A pattern to find the string: "// violation". */
+    /**
+     * A pattern that matches the following comments formats.
+     * <ol>
+     *     <li> "// violation" </li>
+     *     <li> "// violation, 'violation message'" </li>
+     *     <li> "// violation 'violation messages'" </li>
```

Comment:
can we not allow `,` easily ?
better to not have much diversity/freedom in such technical comments.

we will use `,` in cases  when explanation is required, so it will be ` , explanation, `.

Reply:
Why be so restrictive? For BDD, we only care about `violation` and the optional error message regexp enclosed by matching `’`. It is better to make this as easy as possible on contributors. Other aspects of the comment are only for humans, and we have manual review for that.

Reply:
`// violation, 'violation message'`
Or
`// violation 'violation messages'`

Why contributor need to choose between two styles.
`// violation 'violation messages'` is good and without any extra characters.  We do have problems with line length for such lines, the less characters the better.

If we have both already, I completely ok to move update to separate issue. I just curious if the is good reason on extra `,`.

Reply:
> Why contributor need to choose between two styles.

This is my point, why should a contributor have to choose between any styles? BDD should only care about `violation` and violation message content between `’`.  The rest can be left up to humans to make sure it reads well. The comma helps readability. 

This was my point at https://github.com/checkstyle/checkstyle/pull/13502#discussion_r1292824470. We can just combine a bunch of these regexp into one, since the regexp is for BDD and humans review for readability.

Reply:
Still not clear why we should have `,` in regexp at all, better to make such lins less different. But ok  now we can reconsider it later on.